### ISSUE TITLE
timestamps: false schema options does not work for sub-schema

### ISSUE THREAD

**Issue by lykmapipo** [Automattic/mongoose#7074]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
All sub-schema defines using `new Schema([...fields], {timestamps: false})` end up 
having the timestamps
```js
{
 firstName: "John",
 surname: "Doe",
 updatedAt: "2018-10-01T14:05:17.752Z",
 createdAt: "2018-10-01T14:05:17.752Z"
}
```

**If the current behavior is a bug, please provide the steps to reproduce.**
```js
#!/usr/bin/env node

'use strict';

process.env.MONGODB_URI =
    (process.env.MONGODB_URI || 'mongodb://localhost/mongoose-playground');

/* dependencies */
const mongoose = require('mongoose');
const Schema = mongoose.Schema;


mongoose.connect(process.env.MONGODB_URI);

const Name = new Schema({
    firstName: {type: String },
   surname: {type: String }
}, {_id: false, id: false, timestamps: false });

const UserSchema = new Schema({
    name: Name
}, { timestamps: true });

const User = mongoose.model('User', UserSchema);

User.create({ name: {firstName: 'John', surname: 'Doe' } }, function(error, user) {
    console.log(error);
    console.log(user);
});
```

**What is the expected behavior?**
Disabling timestamps i.e `{ timestamps: false }` should remove timestamps `createdAt` and `updatedAt` from `sub-schema` or `sub-doc`

**Please mention your node.js, mongoose and MongoDB version.**
node.js : v10.1.0
mongoose: v5.3.0
MongoDB: v3.4.10

**Comment by gwuah** [comment]
@lineus @vkarpov15  , I'd like to take on this bug.
Can you please guide me on how to fix it?

**Comment by lineus** [comment]
@gwuah I'm not sure if you're asking for procedural steps, like forking the repo in git or which file to start investigating, etc. I would suggest that you start by checking out [the contributing guide](https://github.com/Automattic/mongoose/blob/master/CONTRIBUTING.md) if you haven't already and join us on [Slack](http://slack.mongoosejs.io/) to ask any questions you may have. I look forward to seeing you there 👍

**Comment by lineus** [comment]
@vkarpov15 I think I made a mistake marking this as a bug. [The docs](https://mongoosejs.com/docs/guide.html#timestamps) say 

>If set timestamps, mongoose assigns createdAt and updatedAt fields to your schema, the type assigned is Date.

Looking at the code and the wording in the docs it looks like this is intentional. It's either not set or is set, so true/false appear to have the same effect. Is this something we *should* change or is the answer here to set them to true/obj or don't set them at all?

**Comment by gwuah** [comment]
@lineus 👀

**Comment by Fr4nk1n0** [comment]
**Mongoose version**: 5.3.4
**Node version**: 8.11.3
**MongoDB version**: 3.2.10

**File path**: mongoose/lib/schema.js:853

I think the find condition `s.schema.options.timestamps != null` is to be changed to `s.schema.options.timestamps` without `!= null` because the timestamps of the child Schemas have a Boolean value.

**Extended example**

**From:** 
`const childHasTimestamp = this.childSchemas.find(s => s.schema.options.timestamps != null);`
`if (!timestamps && !childHasTimestamp) {return;}`

**To:**
`const childHasTimestamp = this.childSchemas.find(s => s.schema.options.timestamps);`
`if (!timestamps && !childHasTimestamp) {return;}`

With this change everything started to work again.

**Comment by lineus** [comment]
@Fr4nk1n0 can you show a test case where the current version doesn't behave as expected? I created the following test based on the example code in this issue's original report, but it passes without any changes:

### test added
```js
  it('child schema with explicitly false timestamps works (gh-7074)', function(done) {
    const childSchema = new mongoose.Schema({ name: String }, { timestamps: false });
    const schema = new mongoose.Schema({
      child: childSchema,
      children: [childSchema]
    }, { timestamps: true });

    const M = db.model('gh7074Child', schema);

    M.create({
      child: { name: 'kid1' },
      children: [{ name: 'kid2'}, { name: 'kid3' }]
    }, function(err, doc) {
      assert.ifError(err);
      assert.strictEqual(doc.child.createdAt, undefined);
      assert.strictEqual(doc.child.updatedAt, undefined);
      assert.strictEqual(doc.children[0].createdAt, undefined);
      assert.strictEqual(doc.children[1].updatedAt, undefined);
      done();
    });
  });
```
### Output:
```
mongoose>: git status --porcelain
 M test/timestamps.test.js
mongoose>: npm test -- -g 'gh-7074'

> mongoose@5.3.4 test /Users/lineus/dev/node/src/mongoose
> mocha --exit test/*.test.js test/**/*.test.js "-g" "gh-7074"


 You're not testing shards!
 Please set the MONGOOSE_SHARD_TEST_URI env variable.
 e.g: `mongodb://localhost:27017/database
 Sharding must already be enabled on your database


  ․․

  2 passing (5s)

mongoose>:
```

I'm not refuting that the test in find could be changed, I just need to know how to reproduce the problem it's causing so I can add a test that will prevent this behavior from changing in the future.

**Comment by Fr4nk1n0** [comment]
@lineus Try using **.save()** instead of **.create()**

**Comment by lineus** [comment]
that works for me too:

### updated test
```js
  it('child schema with explicitly false timestamps works (gh-7074)', function(done) {
    const childSchema = new mongoose.Schema({ name: String }, { timestamps: false });
    const schema = new mongoose.Schema({
      child: childSchema,
      children: [childSchema]
    }, { timestamps: true });

    const M = db.model('gh7074Child', schema);
    const m = new M({
      child: { name: 'kid1' },
      children: [{ name: 'kid2' }, { name: 'kid3' }]
    });

    m.save(function(err, doc) {
      assert.ifError(err);
      assert.strictEqual(doc.child.createdAt, undefined);
      assert.strictEqual(doc.child.updatedAt, undefined);
      assert.strictEqual(doc.children[0].createdAt, undefined);
      assert.strictEqual(doc.children[1].updatedAt, undefined);
      done();
    });
  });
```
### Output:
```
mongoose>: npm test -- -g 'gh-7074'

> mongoose@5.3.4 test /Users/lineus/dev/node/src/mongoose
> mocha --exit test/*.test.js test/**/*.test.js "-g" "gh-7074"


 You're not testing shards!
 Please set the MONGOOSE_SHARD_TEST_URI env variable.
 e.g: `mongodb://localhost:27017/database
 Sharding must already be enabled on your database


  ․․

  2 passing (5s)

mongoose>:
```

**Comment by YouriT** [comment]
Here is a different test case that fails the assertion that `timestamps: false` works. By the way in `5.0.12` it was working just fine so not sure when the regression was brought in.

```js
it('timestamps set to false works (gh-7074-bis)', function () {
  const subSchema = new Schema({
    name: String,
    createdAt: {
      type: Date,
      default: Date.now,
      required: true,
    },
  }, { timestamps: false, });
  const schema = new Schema({
    name: String,
    sub: {
      type: subSchema,
      required: false,
    },
  }, {
    timestamps: false, useNestedStrict: true,
  });
  const Test = db.model('gh7074-bis', schema);
  return co(function* () {
    yield Test.deleteMany({});
    const subDoc = { name: 'test2', createdAt: new Date('2018-10-30') };
    const doc = yield new Test({ name: 'test', sub: subDoc }).save();
    assert.strictEqual(doc.updatedAt, undefined);
    assert.strictEqual(doc.createdAt, undefined);
  });
});
```

I can open a new issue if required.

**Comment by vkarpov15** [comment]
Not necessary, will take a closer look at the script later

**Comment by vkarpov15** [comment]
@YouriT this was fixed in 5.3.9 with #7203
